packages/react-components/src/components/Switch.tsx

```diff
@@ -0,0 +1,86 @@
+import * as React from 'react';
+import cx from 'classnames';
+import classNames from 'classnames/bind';
+
+import { noop } from './constants';
+
+const baseClass = 'switch';
+
+enum SwitchSize {
+  Basic = 'basic',
+  Compact = 'compact',
+}
+
+enum SwitchStyles {
+  Enabled = 'enabled',
+  Disabled = 'disabled',
+}
+
+export interface IProps {
+  className?: string;
+  defaultOn?: boolean;
+  disabled?: boolean;
+  name?: string;
+  on?: boolean;
+  onChange?(e: React.FormEvent, value: boolean): void;
+  size?: SwitchSize;
+  innerRef?: React.LegacyRef<HTMLInputElement> | undefined;
+}
+
+export const Switch: React.FC<IProps> = ({
+  className = '',
+  defaultOn = false,
+  disabled = false,
+  name = baseClass,
+  on,
+  onChange = noop,
+  size = SwitchSize.Basic,
+  innerRef,
+  ...props
+}) => {
+  const getEnabledValue = () => (on !== undefined ? on : defaultOn);
+  const [enabled, setEnabled] = React.useState(getEnabledValue());
```

Comment:
The original implementation of `Switch` component contains logic that allows this component to be both: controlled or uncontrolled input making the state being derived from props 👇 

https://github.com/livechat/design-system/blob/0a1035019e20cb84e38d5a03267def2fa9228ee3/packages/design-system/src/components/Switch/Switch.js#L18

Here the controlled input part is missing as there is no synchronization between `on` property and `enabled` state. 
It can be easily added using `useEffect` hook 👇 
```ts
React.useEffect(() => {
    if (on !== undefined) {
      setEnabled(on);
    }
  }, [on]);
```

Reply:
Added, and I have decided to remove `defaultOn` property. Using `on` and `onChange` will cause the same effect